### ISSUE TITLE
core: focus modal/overlay

### ISSUE THREAD

**Issue by coryrylan** [vmware/clarity#5760]
## Describe the bug

When cds-modal is opened the modal content is being focused as part of the focus trap. A11y team would like this removed from the tab order. 

## How to reproduce

1.  Go to https://js-2mnmym.stackblitz.io/
2.  Open modal
3.  Tab through and see content is in tab order

## Expected behavior

Modal content should not be in tab order

## Versions

**Clarity project:**

- [x] Clarity Core
- [ ] Clarity Angular/UI

**Clarity version:**

- [ ] v3.x
- [ ] v4.x
- [x] v5.x

**Framework:**

- [ ] Angular
- [ ] React
- [ ] Vue
- [ ] Other:

**Framework version:**
_ie: Angular 11_

**Device:**

- Type: [e.g. MacBook]

## Additional notes

_Add any other notes about the problem here._

**Comment by mathisscott** [comment]
Per prior discussions on this, our answer will be to drive users towards `cds-first-focus`. That feature will allow end users to customize what will and won't have initial focus when an overlay is opened. `cds-first-focus` removes the overlay wrapper from the tabflow. We recognize, however, that there may be users who _want_ the overlay wrapper in the tab flow. a11y initially requested that to be the case themselves when they advised that the close button no longer receive initial focus.

So rather than force focus onto elements (like a header or close button) that may or may not be there and jump on the shell game of guesstimating use cases, we will be more illustrative of the tools available to users through Core's overlays.

The fix for this issue then will be...

1. Make our overlay and modal demos use first-focus so we set a good example
2. Highlight that the expectation from a11y is that users will use first-focus in their own overlays/modals. This will also enable teams outside of VMware to ignore that if their own a11y teams are telling them something different (e.g. focus the close button, focus the header, focus the wrapper...)

**Comment by Jinnie** [comment]
I remember we had one more reason to have it in the Tab order (at least with the Clarity Angular modals) - be able to scroll the content with keyboard only. 
Is there an internal JIRA issue that covers this new A11Y request? I see one for old Angular modals only (TCS-3639), which refers to the Clarity Known Challenges list, where we have this statement:

> Focus management cannot be granular, therefore focus lands on the container element. Consequently, the screen reader will announce the entire contents of the container element and may seem to announce elements repeatedly with some browser and screen reader combinations (such as NVDA/Firefox). In Clarity v5, focus shifts to the title rather than the container. 

It would be good if we had a Core specific issue where we can track the communication and especially the requirement.

**Comment by mathisscott** [comment]
> I remember we had one more reason to have it in the Tab order (at least with the Clarity Angular modals) - be able to scroll the content with keyboard only.

The content area itself (for modals) is in the tab flow. So that is covered. If someone created a generic overlay, they could do that themselves pretty easily. The content wrapper isn't _removed_ from the tabflow if _cds-first-focus_ is not defined. It also isn't removed from the tab flow if _cds-first-focus_ IS defined. It's just that the first focus element hijacks focus from the contact wrapper. That's why this was such a light fix. It gives more flexibility to our end-users. 

> Is there an internal JIRA issue that covers this new A11Y request? I see one for old Angular modals only (TCS-3639), which refers to the Clarity Known Challenges list, where we have this statement:
> 
> > Focus management cannot be granular, therefore focus lands on the container element. Consequently, the screen reader will announce the entire contents of the container element and may seem to announce elements repeatedly with some browser and screen reader combinations (such as NVDA/Firefox). In Clarity v5, focus shifts to the title rather than the container.

This fix addresses all that.
